src/compiler/checker.ts

```diff
@@ -23301,9 +23305,14 @@ namespace ts {
                 return true;
             }
             const prop = getPropertyOfType(type, propertyName);

```

Comment:
I'm thinking about this `isValidPropertyAccess` function, and I see that it's a public API function that takes the name of a property and a property access node and returns whether the property access is valid. Because it's using `getPropertyOfType`, it seems like this would need to take in our mangled symbol name of the private identifier property. How is this function supposed to be used (wrt to private identifiers)?

Reply:
It's not ideal, but users can grab the properties with `getPropertiesOfType` and `getPropertyOfType`, then get the name via `.name`. 

If there are nice alternatives, I think they're worth considering, though! The problem I see is that a single instance of a class can have multiple private-named properties corresponding to different declarations.

```ts
class A {
    #foo = 3;
    takesOther(b: B) {
         b.#foo
    };
}

class B extends Foo {
     #foo = "I am shadowing the other `#foo`";
}
```